### PR TITLE
Dropdown

### PR DESCRIPTION
**Pull request by nickmccurdy**
## Description
Depends on #63

A dropdown/select/menu toggled by a caret icon

## Tasks
- [x] Add icon
- [x] Custom menu styles (DataList)
- [x] Extract code
- [x] Animation ~(blocked by design spec)~
- [x] Custom content
- [x] Close Dropdown by clicking DropdownButton
- [x] Add DataList's shadow to DropdownButton when open

## [Zeplin Design](https://zpl.io/amdP5A3)

image

## [Storybook Story](https://mycryptobuilds.com/ui/US-38-Dropdown/?selectedKind=Atoms&selectedStory=Dropdown&full=0&addons=1&stories=1&panelRight=0&addonPanel=%40storybook%2Faddon-a11y%2Fpanel)

image

### CODE REVIEW

**Comment by nickmccurdy** [comment]
https://app.clubhouse.io/mycrypto/story/900/ui-create-a-dropdown-component-to-be-used-in-the-select-network-view

**Comment by nickmccurdy** [comment]
https://app.clubhouse.io/mycrypto/story/1192/molecule-dropdown

File: src/molecules/Dropdown/Dropdown.tsx [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+    ) {
+      this.setState({ open: false });
+    }
+  };
+
+  private readonly handleChange = (selected: ReactNode) => () => {
+    const { onChange } = this.props;
+
+    if (onChange) {
+      onChange(selected);
```
[the remark this answers is not in the document]
Comment (nickmccurdy) [reply src/molecules/Dropdown/Dropdown.tsx]: `onChange` now takes a `number` instead

**Comment by vilsbole** [comment]
Can we add some documentation for the component please?
The default storybook is really insufficient. e.g https://mycryptobuilds.com/ui/?selectedKind=Atoms&selectedStory=Heading&full=0&addons=1&stories=1&panelRight=0&addonPanel=%40storybook%2Faddon-a11y%2Fpanel
The complete list of props and the ones that cannot be used together would should be enough

**Comment by vilsbole** [comment]
I don't know if it's been corrected yet but the storybook gif shows that only menu has the overlay shadow while the designs include the input field

**Comment by nickmccurdy** [comment]
> Can we add some documentation for the component please?
> The default storybook is really insufficient. e.g https://mycryptobuilds.com/ui/?selectedKind=Atoms&selectedStory=Heading&full=0&addons=1&stories=1&panelRight=0&addonPanel=%40storybook%2Faddon-a11y%2Fpanel
> The complete list of props and the ones that cannot be used together would should be enough

There are code examples for some of the basic usages in story info. The props table is supposed to list everything but the Storybook plugin that generates the data broke at some point, fixing that is the backlog. Is that enough or should I add doc comments until it's fixed?

**Comment by nickmccurdy** [comment]
> I don't know if it's been corrected yet but the storybook gif shows that only menu has the overlay shadow while the designs include the input field

Sorry forgot to update the gif, it is fixed, now the button raises and is separated from the dropdown by 7px (request from our designer).

**Comment by vilsbole** [comment]
> The props table is supposed to list everything but the Storybook plugin that generates the data broke at some point, fixing that is the backlog. Is that enough or should I add doc comments until it's fixed?

If it's possible to add text to storybook or comments in the code then yes. Especially items vs children

**Comment by nickmccurdy** [comment]
We decided to request time for a doc/tooling overhaul of UI's storybook in the sprint planning meeting.

As far as I'm aware, the technical and design review changes should be met, I just need an additional review to merge.

**Pull request merged**